homeassistant/components/sensor/arwn.py

```diff
@@ -0,0 +1,121 @@
+"""Support for collecting data from the ARWN project.
+
+For more details about this platform, please refer to the
+documentation at https://home-assistant.io/components/sensor.arwn/
+
+"""
+import json
+import logging
+from homeassistant.helpers.entity import Entity
+import homeassistant.components.mqtt as mqtt
+from homeassistant.const import (TEMP_FAHRENHEIT)
+from homeassistant.util import slugify
+
+DEPENDENCIES = ['mqtt']
+
+DOMAIN = "arwn"
+TOPIC = 'arwn/#'
+SENSORS = {}
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def discover_sensors(topic):
+    """Given a topic, dynamically create the right sensor type."""
+    parts = topic.split('/')
+    domain = parts[1]
+    if domain == "temperature":
+        name = parts[2]
+        return (ArwnSensor(name, 'temp', TEMP_FAHRENHEIT),)
```

Comment:
Actually, in my latest iteration I'm publishing units on the wire as well to future proof things. I can rev this to take advantage of that.